### PR TITLE
[8.x] Adds class handling for Blade echo statements

### PR DESCRIPTION
**Pull request by lukeraymonddowning**
## What is it?

When working in Blade, I'll often want to always output a class in the same manner. I'll use Brick/Money as the example here, but this can apply to pretty much any class you can think of. 

Brick/Money is a final class, so I can't add a `Stringable` interface to it. Whenever I output `{{ $moneyObject }}`, I would want it to format as `£100.00`, for example. The same could be said for a `Carbon` instance. 

This PR adds a new `Blade::handle()` method that can be placed in the boot method of a Service Provider and allows the user to add intercepting closures for any class. The returned value will be outputted in Blade.

Think of this as the Exception Handler, but for Blade.

## Usage

```php
// AppServiceProvider
Blade::handle(Money::class, fn($object) => $object->formatTo('en_GB'));
```

```blade
<dl>
    <dt>Total</dt>
    <dd>{{ $total }}</dd> <!-- This is a money object, but will be outputted as an en_GB formatted string -->
</dl>
```

## How does it work?

This works by wrapping a small piece of logic around php output at the parser level, which means that the object is evaluated just in time and can thus be treated as a standard object throughout the entire application; it is only transformed by the PHP script at the time of output.

Feel free to throw any questions back at me :-)

Thanks for all the hard work guys!

### CODE REVIEW

**Comment by driesvints** [comment]
Heya. Seems like this PR breaks the test suite. Feel free to attempt again with passing tests.

**Comment by lukeraymonddowning** [comment]
@driesvints on it now 👍 Its just because it wraps existing output. Will reopen once fixed.

File: src/Illuminate/View/Compilers/BladeCompiler.php [hunk 1]
```diff
@@ -127,6 +127,13 @@ class BladeCompiler extends Compiler implements CompilerInterface
      */
     protected $classComponentNamespaces = [];
 
+    /**
+     * The array of handlers objects should be passed through.
+     *
+     * @var array
+     */
+    public static $echoHandlers = [];
```
Comment (Gummibeer) [inline src/Illuminate/View/Compilers/BladeCompiler.php:135]: Does this have to be a static property? Isn't the `BladeCompiler` registered as singleton? 🤔

**Comment by lukeraymonddowning** [comment]
@driesvints any chance of reopening this? I've fixed the test suite 👍

**Comment by lukeraymonddowning** [comment]
Nevermind, opened a new one #37478

**Pull request closed without merging**